File: 2GII/PDOO/QytetetJava/src/modeloqytetet/Dado.java
package modeloqytetet;

import java.util.concurrent.ThreadLocalRandom;

public class Dado {
    // ----------------------- Constructor -------------------------------------   
    private Dado() { }
    
    // ----------------------- Métodos singleton -------------------------------   
    public static Dado getInstance() {
        return DadoHolder.INSTANCE;
    }
    
    private static class DadoHolder {
        private static final Dado INSTANCE = new Dado();
    }
        
    // ----------------------- Métodos ----------------------------------------   
    // El máximo no es inclusivo, hay que sumarle 1
    int tirar() {                        
        return ThreadLocalRandom.current().nextInt(1,6+1);
    }
    
}

File: 2GII/PDOO/QytetetRuby/lib/dado.rb
#encoding: utf-8

require "singleton"

module ModeloQytetet
  
  class Dado
    include Singleton

    # -------------------------- Constructor -----------------------------------
    def initialize
      # No necesita nada
    end
    
    # ---------------------------- Métodos -------------------------------------
    def tirar
      1 + rand(6)
    end
  end
 
end